schematics/table/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.ts

```diff
@@ -0,0 +1,132 @@
+import { Component, OnInit, ViewChild, <% if(!!viewEncapsulation) { %>, ViewEncapsulation<% }%><% if(changeDetection !== 'Default') { %>, ChangeDetectionStrategy<% }%> } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { map } from 'rxjs/operators/map';
+import { MatPaginator, MatSort } from '@angular/material';
+import { DataSource } from '@angular/cdk/collections';
+import { of } from 'rxjs/observable/of';
+import { merge } from 'rxjs/observable/merge';
+
+export interface Element {
+  name: string;
+  position: number;
+  weight: number;
+  symbol: string;
+}
+
+export class ExampleDataSource extends DataSource<any> {
```

Comment:
The data source should go in its own file. I'd like this data source to play a big role in guiding users in structuring their data access and manipulation.

It would be great if we could add an option for a client data-source vs. a server data source. The client one would perform everything locally and synchronously, while the server one would call out to (fake) server methods and do thing asynchronously.